### ISSUE TITLE
SQLite backend contains bad timestamp

### ISSUE THREAD

**Issue by Vexs** [JWCook/aiohttp-client-cache#77]
### The problem
I receive the following traceback when accessing the following URL `https://www.bizjournals.com/albuquerque/news/2021/07/01/bayotech-receives-investment-from-caterpillar.html`
The URL hits us with an anti-bot, but that's not the concern, just an unusual traceback. 
```
Traceback (most recent call last):
  File "C:/Users/james/PycharmProjects/automation/Automation_backend/code/utils/nlp/downloader/article_downloader.py", line 194, in worker
    articles_data.append(await download(session, new_link))
  File "C:/Users/james/PycharmProjects/automation/Automation_backend/code/utils/nlp/downloader/article_downloader.py", line 64, in download
    async with session.get(link, timeout=60) as resp:
  File "C:\Users\james\PycharmProjects\automation\venv\lib\site-packages\aiohttp\client.py", line 1117, in __aenter__
    self._resp = await self._coro
  File "C:\Users\james\PycharmProjects\automation\venv\lib\site-packages\forge\_revision.py", line 322, in inner
    return await callable(*mapped.args, **mapped.kwargs)
  File "C:\Users\james\PycharmProjects\automation\venv\lib\site-packages\aiohttp_client_cache\session.py", line 28, in _request
    response, actions = await self.cache.request(method=method, url=str_or_url, **kwargs)
  File "C:\Users\james\PycharmProjects\automation\venv\lib\site-packages\aiohttp_client_cache\backends\base.py", line 117, in request
    response = None if actions.skip_read else await self.get_response(actions.key)
  File "C:\Users\james\PycharmProjects\automation\venv\lib\site-packages\aiohttp_client_cache\backends\base.py", line 132, in get_response
    elif not self.is_cacheable(response):  # type: ignore
  File "C:\Users\james\PycharmProjects\automation\venv\lib\site-packages\aiohttp_client_cache\backends\base.py", line 83, in is_cacheable
    'not expired': not getattr(response, 'is_expired', False),
  File "C:\Users\james\PycharmProjects\automation\venv\lib\site-packages\aiohttp_client_cache\response.py", line 133, in is_expired
    return self.expires is not None and datetime.utcnow() > self.expires
TypeError: can't compare offset-naive and offset-aware datetimes
```

Best guess? Something funny got saved, because I can't recreate this error with a new sqlite db file. 

### Expected behavior
For the website to tell my script to bugger off

### Steps to reproduce the behavior
_With a complete code example, if possible_

```
url = "https://www.bizjournals.com/albuquerque/news/2021/07/01/bayotech-receives-investment-from-caterpillar.html"
session = CachedSession(cache=SQLiteBackend("request_cache.sqlite"))
async with session.get(url) as r:
    print(await r.text())
```
And you'll need the following DB file, which I've stripped out sans the odd stored value.
https://mega.nz/file/LXhEhaxB#2i4Wx3qLB2QuFSOHNF9IdQESsyxIGxNfy30n8-WVWTE

### Workarounds
Manually clearing the bad entry

### Environment
- aiohttp-client-cache version: 0.4.2
- Python version: 3.8.5
- Platform: win10

Thanks for resolving #75 so quickly by the way.

**Comment by JWCook** [comment]
I believe that cached response must have been created with aiohttp-client-cache 0.3 or older. I fixed the datetime/timezone comparison issue here: [eecdcef](https://github.com/JWCook/aiohttp-client-cache/commit/eecdcef7e5e2cc8ed8746532ab8141ca17da7386#diff-8cb15b25b049088fec79bbb9c8c6f96e604eb12984f9fa832dfec51657232ff2R203-R210)

In general, the cache format may change slightly with each release, which means older cached responses will become invalid after updating, but the intended behavior is to quietly fail and re-fetch the response. This particular error didn't get caught since it happened _after_ deserialization.

The timezone issue won't be a problem from 0.4.0 on. Do you think it's worth adding error handling for this, or are you okay with just clearing your cache?

**Comment by Vexs** [comment]
Hm, this backend was created yesterday on the 0.4.1 version I believe- unless 0.3 was up on pypi yesterday.

**Comment by JWCook** [comment]
Oh, interesting. I'm not sure why that would be happening, then. Let me take a closer look at that cache file...

**Comment by JWCook** [comment]
That cached response does have a timezone-aware datetime:
```python
datetime.datetime(2020, 7, 25, 20, 18, 46, tzinfo=datetime.timezone.utc)
```

I still need to figure out how it got there, that's a tricky one.

**Comment by Vexs** [comment]
Huh, check the year- that's weird, right?

**Comment by JWCook** [comment]
Aha! Found it. So with that original cached response, somehow you didn't get the anti-bot response and got the actual site content, along with these headers:
```python
{'Cache-Control': 'no-cache, must-revalidate',
 'Connection': 'keep-alive',
 'Content-Encoding': 'gzip',
 'Content-Security-Policy': "frame-ancestors 'self' media.bizj.us *.vwo.com "
                            'visualwebsiteoptimizer.com '
                            'tpc.googlesyndication.com; '
                            'upgrade-insecure-requests;',
 'Content-Type': 'text/html; charset=UTF-8',
 'Date': 'Mon, 26 Jul 2021 00:18:46 GMT',
 'Expires': 'Sat, 25 Jul 2020 20:18:46 GMT',
 'Feature-Policy': "accelerometer 'self' https://tpc.googlesyndication.com "
                   'https://z.moatads.com https://px.moatads.com '
                   'https://geo.moatads.com https://mb.moatads.com',
 'Server': 'Apache',
 'ServerID': 'production',
 'Set-Cookie': 'bizj=deleted; expires=Thu, 01-Jan-1970 00:00:01 GMT; '
               'Max-Age=0; path=/; domain=.www.bizjournals.com',
 'Strict-Transport-Security': 'max-age=31536000',
 'Transfer-Encoding': 'chunked',
 'Vary': 'Accept-Encoding',
 'X-CDN': 'Imperva',
 'X-Iinfo': '14-10338238-10338242 NNNN CT(38 72 0) RT(1627258726365 36) q(0 0 '
            '1 -1) r(6 6) U12',
 'X-UIN': 'lxE7CfNxIw1Yzf8MGbOCmA08379108'}
```

That 'Expires' value then got used as the expiration time (in [cache_control.py#L116](https://github.com/JWCook/aiohttp-client-cache/blob/main/aiohttp_client_cache/cache_control.py#L116)). So there are actually 2 bugs here:
* Response headers aren't supposed to be used unless you specify `cache_control=True`
* Datetimes parsed from headers should be converted to UTC

...And yes, it is weird to have an expiration date over a year old. Someone's web server has some major clock drift going on!

**Comment by Vexs** [comment]
Nice catch! That's rather obtuse for sure

**Comment by JWCook** [comment]
This is now fixed in `0.4.3`.
